.github/workflows/publish-release.yml

```
@@ -33,20 +33,23 @@ jobs:
       - uses: actions/cache@v3
         id: restore-build
         with:
-          path: |
-            ./*
-            !./node_modules/
+          path: ./dist
           key: ${{ github.sha }}
 
   publish-npm:
     environment: publish
     runs-on: ubuntu-latest
     needs: publish-release
     steps:
+      - uses: actions/checkout@v2
+        with:
+          # We check out the release pull request's base branch, which will be
+          # used as the base branch for all git operations.
+          ref: ${{ github.event.pull_request.base.ref }}
```

If you want to be super pedantic about it, passing the exact commit hash from one step to another prevents you from publishing stuff in dist along with mismatched package.json coming from a force-push

---

I don't think we need to explicitly pass the commit hash either. It's included in the `github.event.pull_request` event. We can reference it instead of `.ref`.

```suggestion
          ref: ${{ github.event.pull_request.base.sha }}
```

Property name taken from this example in docs: https://docs.github.com/en/developers/webhooks-and-events/webhooks/webhook-events-and-payloads#webhook-payload-example-32